File: src/Api/Model/DB/NaturezaJuridica.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.Text;

namespace CriadorBaseDados.Model.DB
{
    [JsonObject(MemberSerialization.OptIn)]
    public class NaturezaJuridica : RealmObject
    {
        [JsonProperty]
        [PrimaryKey]
        public int Cod { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodGrupo => Grupo.Cod;
        public GrupoNaturezaJuridica Grupo { get; set; }
        [JsonProperty]
        public string Descricao { get; set; }
    }
}

File: src/Mobile/b2b/lib/model/core/natureza_juridica.dart
class NaturezaJuridica {
  final int cod;
  final int codGrupo;
  final String descricao;

  NaturezaJuridica(this.cod, this.codGrupo, this.descricao);

  NaturezaJuridica.fromJson(Map<String, dynamic> json)
      : cod = json['cod'],
        codGrupo = json['codGrupo'],
        descricao = json['descricao'];

  Map<String, dynamic> toMap() => <String, dynamic>{
        'cod': cod,
        'codGrupo': codGrupo,
        'descricao': descricao,
      };
}
